lib/Doctrine/Common/Annotations/AnnotationReader.php

```diff
@@ -301,41 +302,82 @@ public function getMethodAnnotation(ReflectionMethod $method, $annotationName)
     }
 
     /**
-     * Returns the ignored annotations for the given class.
+     * Gets the annotations applied to a function.
      *
-     * @param \ReflectionClass $class
+     * @param ReflectionFunction $function The ReflectionFunction of the function
+     *                                     from which the annotations should be read.
      *
-     * @return array
+     * @return array An array of Annotations.
      */
-    private function getIgnoredAnnotationNames(ReflectionClass $class)
+    public function getFunctionAnnotations(ReflectionFunction $function) : array
     {
-        $name = $class->getName();
-        if (isset($this->ignoredAnnotationNames[$name])) {
-            return $this->ignoredAnnotationNames[$name];
-        }
+        $context = 'function ' . $function->getName();
 
-        $this->collectParsingMetadata($class);
+        $this->parser->setTarget(Target::TARGET_FUNCTION);
+        $this->parser->setImports($this->getImports($function));
+        $this->parser->setIgnoredAnnotationNames($this->getIgnoredAnnotationNames($function));
+        $this->parser->setIgnoredAnnotationNamespaces(self::$globalIgnoredNamespaces);
 
-        return $this->ignoredAnnotationNames[$name];
+        return $this->parser->parse($function->getDocComment(), $context);
     }
 
     /**
-     * Retrieves imports.
+     * Gets a function annotation.
      *
-     * @param \ReflectionClass $class
+     * @param ReflectionFunction $function       The ReflectionFunction to read the annotations from.
+     * @param string             $annotationName The name of the annotation.
```

Comment:
I'd remove that line, and maybe also the line before that.

Reply:
I've added them for consistency with other methods : https://github.com/doctrine/annotations/blob/dc296632db0500c7bd5144161cf96708f61059ff/lib/Doctrine/Common/Annotations/Reader.php#L42-L44

But it's true that they don't bring much value.

Reply:
Then please don't add them, and hopefully the next contributor will try to be consistent with what _you_ did :crossed_fingers:

Reply:
Done.